bliss/encoder/encoder.py

```diff
@@ -227,27 +227,37 @@ def training_step(self, batch, batch_idx, optimizer_idx=0):
         """Training step (pytorch lightning)."""
         return self._compute_loss(batch, "train")
 
-    def update_metrics(self, batch, batch_idx):
+    def sample_for_metrics(self, batch):
```

Comment:
This new function makes the encoder somewhat harder to understand, in part because it seems arbitrary what stays in `update_metrics` and what goes to `sample_for_metrics`. Probably better to duplicate a bit of code in the notebooks rather than make this change.